semantic_conventions/trace/messaging.yaml

```
@@ -260,3 +260,66 @@ groups:
               brief: 'Broadcasting consumption model'
         brief: >
           Model of message consumption. This only applies to consumer spans.
+
+  - id: messaging.pulsar
+    prefix: messaging.pulsar
+    extends: messaging
+    brief: >
+      Attributes for Apache Pulsar
+    attributes:
+      - id: subscription
+        type: string
+        brief: >
+          Subscription name of Pulsar consumer.
+        example: 'mySubscription'
+      - id: subscription_mode
+        type:
+          allow_custom_values: false
+          members:
+            - id: Durable
+              value: 'Durable'
+              brief: 'A subscription backed by a durable cursor that will retain messages and persist the
+              current position'
+            - id: NonDurable
+              value: 'NonDurable'
+              brief: 'A lightweight subscription mode that does not have a durable cursor associated'
+        brief: >
+          Types of subscription mode supported by Pulsar.
```

```suggestion
          Subscription modes supported by Pulsar
```

Also maybe add a link to the official docs? (might be tricky if they change it since it says "next" for version) https://pulsar.apache.org/docs/next/concepts-messaging#subscription-modes